src/helpers/mod.rs

```diff
@@ -209,12 +209,9 @@ mod test {
             let v = h.param(0).unwrap();
 
             if !h.is_block() {
-                let output = format!("{}:{}", h.name(), v.value().render());
-                out.write(output.as_ref())?;
+                write!(out, "{}:{}", h.name(), v.value().render())?;
             } else {
-                let output = format!("{}:{}", h.name(), v.value().render());
```

Comment:
This one may be even smarter and move out of the if `write!(out, "{}:{}", h.name(), v.value().render())?` and just have the iff for 
```
out.write("->")?;
h.template().unwrap().render(r, ctx, rc, out)?;
```

Reply:
(it's a detail and must not block IHMO it don't hurt to stay as it is.)

Reply:
Right. This can be simplified.